src/custom/components/Web3Status/index.tsx

```diff
@@ -1,36 +1,32 @@
-import React, { useMemo } from 'react'
+import React from 'react'
 import { useWeb3React } from '@web3-react/core'
 import useENSName from 'hooks/useENSName'
-import { isTransactionRecent, useAllTransactions } from 'state/transactions/hooks'
 import { NetworkContextName } from 'constants/index'
 
-import WalletModal from '@src/components/WalletModal'
-import { Web3StatusInner, newTransactionsFirst } from './Web3StatusMod'
+import WalletModal from 'components/WalletModal'
+import { Web3StatusInner } from './Web3StatusMod'
+import useAllTransactionsAndOrders from 'custom/hooks/useAllTransactionsAndOrders'
 
 export default function Web3Status() {
   const { active, account } = useWeb3React()
   const contextNetwork = useWeb3React(NetworkContextName)
 
   const { ENSName } = useENSName(account ?? undefined)
 
-  // ---------- TODO: get tx and meta-tx
-  const allTransactions = useAllTransactions()
-  const sortedRecentTransactions = useMemo(() => {
-    const txs = Object.values(allTransactions)
-    return txs.filter(isTransactionRecent).sort(newTransactionsFirst)
-  }, [allTransactions])
-  const pending = sortedRecentTransactions.filter(tx => !tx.receipt).map(tx => tx.hash)
-  const confirmed = sortedRecentTransactions.filter(tx => tx.receipt).map(tx => tx.hash)
-  // ----------------------------
+  const { pendingTransactions, confirmedTransactions } = useAllTransactionsAndOrders()
 
   if (!contextNetwork.active && !active) {
     return null
   }
 
   return (
     <>
-      <Web3StatusInner pendingCount={pending.length} />
-      <WalletModal ENSName={ENSName ?? undefined} pendingTransactions={pending} confirmedTransactions={confirmed} />
+      <Web3StatusInner pendingCount={pendingTransactions.concat(confirmedTransactions).length} />
```

Comment:
```suggestion
      <Web3StatusInner pendingCount={pendingTransactions.length + confirmedTransactions.length} />
```

Don't concat, array manipulation is needlessly expensive when simple arithmetics would suffice

Reply:
yeah i had this before, need to check that arrays can't be falsy